File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/service/formatter/MonthValueFormatter.java
package io.github.zwieback.familyfinance.business.chart.service.formatter;

import com.github.mikephil.charting.formatter.ValueFormatter;

import org.threeten.bp.LocalDate;

import io.github.zwieback.familyfinance.util.DateUtils;

public class MonthValueFormatter extends ValueFormatter {

    @Override
    public String getFormattedValue(float monthsFromEpoch) {
        LocalDate month = determineCorrectMonth(monthsFromEpoch);
        return DateUtils.localDateToString(month, DateUtils.getISO_LOCAL_MONTH());
    }

    private static LocalDate determineCorrectMonth(float monthsFromEpoch) {
        return DateUtils.epochMonthToLocalDate((long) monthsFromEpoch);
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/service/formatter/MonthValueFormatter.kt
package io.github.zwieback.familyfinance.business.chart.service.formatter

import io.github.zwieback.familyfinance.constant.DateConstants
import io.github.zwieback.familyfinance.extension.MonthsFromEpoch
import org.threeten.bp.format.DateTimeFormatter

class MonthValueFormatter : SpecificPeriodFormatter() {

    override fun getFormatter(): DateTimeFormatter = DateConstants.ISO_LOCAL_MONTH

    override fun getSpecificPeriod(value: Long) = MonthsFromEpoch(value)
}
